udpa/annotations/sensitive.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package udpa.annotations;
+
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FieldOptions {
+  // Magic number derived from concatenating the 7-bit (lower ASCII) characters
+  // 0x4E ('N'), 0x53 ('S'), 0x46 ('F'), 0x57 ('W').
+  SensitiveAnnotation sensitive = 164946775;
+}
+
+message SensitiveAnnotation {
+  // When set to true, `redact` indicates that sensitive data, such as personally identifiable
+  // information, passwords, private keys, etc. should be sanitized for display by tools aware
+  // of this annotation. Note that that this has no effect on standard Protobuf functions such
+  // as `TextFormat::PrintToString`.
+  bool redact = 1;
```

Comment:
This is going to look more like `[(udpa.annotations.sensitive).redact = true`. Do we want to just put this as the field option, or will other things be added here later?

Reply:
In my original envoyproxy PR, I didn't have a separate message, it was just `[(envoy.protobuf.redacted) = true]`. But I think the convention used by the other annotations here is a good idea for the sake of futureproofing.